lxd/daemon.go

```diff
@@ -260,6 +261,48 @@ func allowPermission(objectType auth.ObjectType, entitlement auth.Entitlement, m
 	}
 }
 
+// rbacPermission must be used as an access handler only when permissions that apply to RBAC/TLS drivers do not apply to
+// other drivers. If another driver is loaded, this will always return response.EmptySyncResponse.
+func rbacPermission(permission auth.Permission) func(d *Daemon, r *http.Request) response.Response {
```

Comment:
@markylaing could you explain the differences that you couldn't resolve between rbac and openfga please? Thanks

Reply:
This is to maintain the behaviour of TLS and RBAC drivers when we list API resources. On say `GET /1.0/instances` the TLS and RBAC drivers will return `403 Forbidden` if the user does not have `view` permission on the project (and if all-projects is used, the user must be an admin). Whereas with OpenFGA we will filter out all instances that the user cannot view, likely returning an empty list.

Reply:
https://github.com/canonical/lxd/pull/12313#discussion_r1336893508